overdrive.py

```diff
@@ -525,11 +525,19 @@ def metadata_lookup_obj(self, identifier):
     def make_link_safe(self, url):
         """Turn a server-provided link into a link the server will accept!
 
-        This is completely obnoxious and I have complained about it to
+        The {} part is completely obnoxious and I have complained about it to
         Overdrive.
+
+        The availability part is to make sure we always use v2 of the
+        availability API, even if Overdrive sent us a link to v1.
         """
         parts = list(urlparse.urlsplit(url))
         parts[2] = urllib.quote(parts[2])
+        if (parts[2].startswith("/v1/collections/")
+            and parts[2].endswith("/availability")):
```

Comment:
Is there any chance the Overdrive URL can end with `/availability/` (with an additional `/` at the end)?

Reply:
There are no examples like this in the API documentation, but it's easy to add a little defensive code.

The URL might have a query string, but urlsplit takes care of that.